src/ripple/app/tx/impl/ApplyContext.cpp

```diff
@@ -69,4 +71,54 @@ ApplyContext::visit (std::function <void (
     view_->visit(base_, func);
 }
 
+template<std::size_t... Is>
+TER
+ApplyContext::checkInvariantsHelper(TER terResult, std::index_sequence<Is...>)
+{
+    if (view_->rules().enabled(featureEnforceInvariants) &&
+        app.config().ENFORCE_INVARIANTS)
```

Comment:
The more I look at it, the less I like `ENFORCE_INVARIANTS`, particularly in light of https://github.com/ripple/rippled/commit/f5af8b03de8d770d49e444b60ffe2b9063f0ad91, which moved us a step toward removing `Application`/`Config` dependencies from a lot of calls.

If the ability to disable invariant checking at run-time is really necessary, how about defining a `NoEnforceInvariants` amendment which is *NOT* added to `supportedAmendments`, but can be added to `[features]` without needing any new config code?